geode-redis/src/distributedTest/java/org/apache/geode/redis/RedisSessionDistDUnitTest.java

```diff
@@ -0,0 +1,395 @@
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.redis;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.Serializable;
+import java.net.HttpCookie;
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.springframework.boot.SpringApplication;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.web.client.RestTemplate;
+import redis.clients.jedis.Jedis;
+
+import org.apache.geode.internal.AvailablePortHelper;
+import org.apache.geode.redis.springRedisTestApplication.RedisSpringTestApplication;
+import org.apache.geode.test.awaitility.GeodeAwaitility;
+import org.apache.geode.test.dunit.VM;
+import org.apache.geode.test.dunit.rules.ClusterStartupRule;
+import org.apache.geode.test.dunit.rules.DistributedRestoreSystemProperties;
+import org.apache.geode.test.dunit.rules.MemberVM;
+import org.apache.geode.test.junit.categories.RedisTest;
+
+@Category({RedisTest.class})
+public class RedisSessionDistDUnitTest implements Serializable {
+
+  @ClassRule
+  public static ClusterStartupRule cluster =
+      new ClusterStartupRule(5);
+
+  @Rule
+  public DistributedRestoreSystemProperties restoreSystemProperties =
+      new DistributedRestoreSystemProperties();
+
+  public static ConfigurableApplicationContext springApplicationContext;
+
+  private String LOCALHOST = "http://127.0.0.1";
+
+  private static int indexOfLocator = 0;
+  private static int indexOfServer1 = 1;
+  private static int indexOfServer2 = 2;
+  private static int indexOfClient1 = 3;
+  private static int indexOfClient2 = 4;
+
+  private static VM client1;
+  private static VM client2;
+
+  private static int server1Port;
+  private static int server2Port;
+  private static int client1Port;
+  private static int client2Port;
+
+  private static Properties redisPropsForServer1;
+  private static Properties redisPropsForServer2;
+  private static MemberVM locator;
+
+  private static RestTemplate restTemplate;
+
+  private static Jedis jedis;
+
+  private static final int JEDIS_TIMEOUT =
+      Math.toIntExact(GeodeAwaitility.getTimeout().toMillis());
+
+  @BeforeClass
+  public static void setup() {
+
+    int[] availablePorts =
+        AvailablePortHelper.getRandomAvailableTCPPorts(4);
+
+    server1Port = availablePorts[0];
+    server2Port = availablePorts[1];
+    client1Port = availablePorts[2];
```

Comment:
Since these have the same values and almost the same names, do they really buy any clarity? It seems to make things more confusing down below when they are used.